pkg/util/grpcencoding/s2/s2.go

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/mostynb/go-grpc-compression/blob/f7e92b39057ca421a6485f650243a3e804036498/internal/s2/s2.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: Copyright 2022 Mostyn Bramley-Moore.
+
+// Package s2 is an experimental wrapper for using
+// github.com/klauspost/compress/s2 stream compression with gRPC.
+package s2
```

Comment:
is there a test file we can copy too? :)

Reply:
I have one - https://github.com/deniszh/dskit/blob/7825854f6a7b8baae00dfd970b9810eb424848da/grpcencoding/s2/s2_test.go - but it uses testify, maybe need to be adopted if don't want to bring testify to deps

Reply:
@deniszh Mimir already depends on testify.

Reply:
I can try to copy the test.

Reply:
@deniszh license wise, are you OK with my simply copying your [test code](https://github.com/deniszh/dskit/blob/7825854f6a7b8baae00dfd970b9810eb424848da/grpcencoding/s2/s2_test.go) into Mimir without a notice of your copyright?

Reply:
Otherwise, I could include a `Provenance-includes-*` block.

Reply:
I pushed it to dskit and signed CLA. And I also borrowed it partially from https://github.com/grafana/dskit/blob/main/grpcencoding/snappy/snappy_test.go